specification/wse/SPEC.md

```diff
@@ -635,7 +644,37 @@ because each character is represented by a single byte and exactly 256 distinct
 character codes match their byte representations when processed by the client.  The remaining character codes can be mapped to
 the corresponding binary representations accordingly.
 
-### Binary as Escaped Text
+IE8+ `XDomainRequest` can distinguish all 256 byte-as-character-code values, for a downstream response body with content-type 
+`text/plain;charset=windows-1252`.
+
+However, the `XDomainRequest` `POST` request cannot specify content-type, and so `text/plain;charset=UTF-8` is assumed, but a bug remains where the `\0` (NUL) character cannot be included in the `POST` body since the text is then clipped at the `\0`.
+
+Rather than escaping the `\0` (NUL) byte-as-character-code, the client MAY use character code `256` to represent zero, so the
+client MAY send a `\u0100` character for each `\u0000` character.
+
+The server MUST decode the binary-as-mixed-text upstream such that each character codepoint is computed `mod 0x0100` to determine each originally intended byte value.
+
+The upstream HTTP request MUST have content type `text/plain;charset=utf-8`. The upstream request body MUST contain one or more valid emulated WebSocket frames encoded into valid UTF-8 by transforming each byte as follows:
+
+* If the leading (most significant) bit is set, then the byte MUST the encoded as two bytes, in the form `1100 00?? 10?? ????` where `?` represents one bit of the original byte
+* If the leading bit is not set, then the byte is encoded as is
+* If the byte value is 0x00 it MAY be encoded as 0xC4 0x80 (instead of 0x00)
```

Comment:
I think it's 0xC4 0x80 because that is
`‭‭1100 0100 1000 0000‬'
which is UTF-8 codepoint binary 001 0000 0000 (last 5 bites of first byte plus last 6 of second) which is denary 256. See https://en.wikipedia.org/wiki/UTF-8. Does that make sense or am I misunderstanding something?